src/fuselage/CCPACSSkinSegment.cpp

```diff
@@ -0,0 +1,169 @@
+    const gp_Pnt p6(p5.X(), yMid, p5.Z());
+
+    const gp_Ax1 ref1(p1, gp_Vec(p1, p3));
+    const gp_Ax1 ref2(p2, gp_Vec(p2, p4));
+    const gp_Ax1 ref3(p3, gp_Vec(p3, p1));
+    const gp_Ax1 ref4(p4, gp_Vec(p4, p2));
+    const gp_Ax1 ref5(p6, gp_Vec(p6, p5));
+
+    if (gp_Ax1(p1, gp_Vec(p1, point)).Angle(ref1) >= angleUpper) return false;
+    if (gp_Ax1(p2, gp_Vec(p2, point)).Angle(ref2) >= angleUpper) return false;
+    if (gp_Ax1(p3, gp_Vec(p3, point)).Angle(ref3) >= angleUpper) return false;
+    if (gp_Ax1(p4, gp_Vec(p4, point)).Angle(ref4) >= angleUpper) return false;
+
+    double maxAngle = 0;
+    maxAngle = std::max(maxAngle, gp_Ax1(p6, gp_Vec(p6, c.sFrame_sStringer.Location())).Angle(ref5));
+    maxAngle = std::max(maxAngle, gp_Ax1(p6, gp_Vec(p6, c.eFrame_sStringer.Location())).Angle(ref5));
+    maxAngle = std::max(maxAngle, gp_Ax1(p6, gp_Vec(p6, c.sFrame_eStringer.Location())).Angle(ref5));
+    maxAngle = std::max(maxAngle, gp_Ax1(p6, gp_Vec(p6, c.eFrame_eStringer.Location())).Angle(ref5));
+
+    return gp_Ax1(p6, gp_Vec(p6, point)).Angle(ref5) < maxAngle;
+}
+
+void CCPACSSkinSegment::UpdateBorders()
+{
+    try {
+        CCPACSFrame& sFrame               = m_uidMgr->ResolveObject<CCPACSFrame>(m_startFrameUID);
+        CCPACSFrame& eFrame               = m_uidMgr->ResolveObject<CCPACSFrame>(m_endFrameUID);
+        CCPACSFuselageStringer& sStringer = m_uidMgr->ResolveObject<CCPACSFuselageStringer>(m_startStringerUID);
+        CCPACSFuselageStringer& eStringer = m_uidMgr->ResolveObject<CCPACSFuselageStringer>(m_endStringerUID.value());
+
+        m_borderCache  = BorderCache();
+        BorderCache& c = *m_borderCache;
+
+        // get the intersection points between stringer and frames
+        UpdateBorder(c.sFrame_sStringer, sFrame.GetGeometry(true), sStringer.GetGeometry(true));
+        UpdateBorder(c.sFrame_eStringer, sFrame.GetGeometry(true), eStringer.GetGeometry(true));
+        UpdateBorder(c.eFrame_sStringer, eFrame.GetGeometry(true), sStringer.GetGeometry(true));
+        UpdateBorder(c.eFrame_eStringer, eFrame.GetGeometry(true), eStringer.GetGeometry(true));
+
+        // generate directions to the opposite intersection point
+        c.sFrame_sStringer.SetDirection(gp_Vec(c.sFrame_sStringer.Location(), c.eFrame_eStringer.Location()));
+        c.sFrame_eStringer.SetDirection(gp_Vec(c.sFrame_eStringer.Location(), c.eFrame_sStringer.Location()));
+        c.eFrame_sStringer.SetDirection(gp_Vec(c.eFrame_sStringer.Location(), c.sFrame_eStringer.Location()));
+        c.eFrame_eStringer.SetDirection(gp_Vec(c.eFrame_eStringer.Location(), c.sFrame_sStringer.Location()));
+    }
+    catch (...) {
+        m_borderCache = boost::none;
+    }
+}
+
+void CCPACSSkinSegment::UpdateBorder(gp_Ax1& b, TopoDS_Shape s1, TopoDS_Shape s2)
+{
+    b.SetLocation(GetIntersectionPoint(s1, s2));
+}
+
+gp_Pnt CCPACSSkinSegment::GetIntersectionPoint(TopoDS_Shape s1, TopoDS_Shape s2) const
```

Comment:
To avoid code duplication, please replace this function with https://github.com/DLR-SC/tigl/blob/1b7605e835b1b07e26789934c2077b7b8dd84392/src/common/tiglcommonfunctions.h#L123